lib/segment/src/index/hnsw_index/point_filterer.rs

```diff
@@ -0,0 +1,79 @@
+use bitvec::slice::BitSlice;
+use common::types::{PointOffsetType, ScoredPointOffset};
+
+use crate::id_tracker::IdTracker;
+use crate::payload_storage::FilterContext;
+use crate::vector_storage::{VectorStorage, check_deleted_condition};
+
+#[derive(Clone, Copy)]
+pub struct SimplePointsFilterer<'a> {
```

Comment:
I think this name would define it's purpose a bit better:

```suggestion
pub struct DeletedPointsFilter<'a> {
```

Simple implies to me that it is a naive filter, similar to all the other 'simple's we use across our code base.